src/Neo/ProtocolSettings.cs

```diff
@@ -220,5 +222,37 @@ private static void CheckingHardfork(ProtocolSettings settings)
                 }
             }
         }
+
+        /// <summary>
+        /// Check if the Hardfork is Enabled
+        /// </summary>
+        /// <param name="hardfork">Hardfork</param>
+        /// <param name="index">Block index</param>
+        /// <returns>True if enabled</returns>
+        public bool IsHardforkEnabled(Hardfork hardfork, uint index)
+        {
+            // Return true if there's no specific configuration
+            if (Hardforks.Count == 0)
+                return true;
+
+            // If the hardfork isn't specified in the configuration, check if it's a new one.
+            if (!Hardforks.ContainsKey(hardfork))
+            {
+                int currentHardforkIndex = AllHardforks.IndexOf(hardfork);
+                int lastConfiguredHardforkIndex = AllHardforks.IndexOf(Hardforks.Keys.Last());
+
+                // If it's a newer hardfork compared to the ones in the configuration, disable it.
+                if (currentHardforkIndex > lastConfiguredHardforkIndex)
+                    return false;
+            }
+
+            if (Hardforks.TryGetValue(hardfork, out uint height))
```

Comment:
Variable 'Hardforks' are the enabled ones, right?
Maybe change to ActiveHardforks

Reply:
All hardforks are supposed to be actived, isn't it? otherwise why we put a hardfork there.

Reply:
What is this variable `Hardfork`?

Reply:
Hardforks are the forks that is in the configuration file, while AllHardforks contains the hardforks defined in the enum.